client/signup/steps/clone-credentials/index.jsx

```diff
@@ -59,10 +55,8 @@ class CloneCredentialsStep extends Component {
 		);
 	}
 
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( 'success' === nextProps.updateStatus && ! this.state.gotSuccess ) {
-			this.setState( { gotSuccess: true } );
+	componentDidUpdate( prevProps ) {
+		if ( 'success' !== prevProps.updateStatus && 'success' === this.props.updateStatus ) {
```

Comment:
Is there any possibility for this to happen twice during the component's lifecycle? Theoretically, it's possible, but I couldn't reproduce that kind of scenario TBH, was just curious if you tried it.

Reply:
Not sure. I think we use this kind of pattern in a few places and I haven't seen any problems thus far. Do you think we should add a safeguard to make sure we don't call `goToNextStep` twice?

Reply:
I agree with you, I think we're good to go as it is.